Doc/c-api/init.rst

```diff
@@ -7,6 +7,227 @@
 Initialization, Finalization, and Threads
 *****************************************
 
+.. _pre-init-safe:
+
+Before Python Initialization
+============================
+
+In an application embedding  Python, the :c:func:`Py_Initialize` function must
+be called before using any other Python/C API functions; with the exception of
+a few functions and the :ref:`global configuration variables <global-conf-vars>`:
+
+The following functions can be safetely called before Python is initialized:
+
+* Configuration functions:
+
+  * :c:func:`PyImport_AppendInittab`
+  * :c:func:`PyImport_ExtendInittab`
+  * :c:func:`PyInitFrozenExtensions`
+  * :c:func:`PyMem_GetAllocator`
+  * :c:func:`PyMem_SetAllocator`
+  * :c:func:`PyMem_SetupDebugHooks`
+  * :c:func:`PyObject_GetArenaAllocator`
+  * :c:func:`PyObject_SetArenaAllocator`
+  * :c:func:`Py_GetBuildInfo`
+  * :c:func:`Py_GetCompiler`
+  * :c:func:`Py_GetCopyright`
+  * :c:func:`Py_GetPlatform`
+  * :c:func:`Py_GetProgramName`
+  * :c:func:`Py_GetVersion`
+  * :c:func:`Py_SetPath`
+  * :c:func:`Py_SetProgramName`
+  * :c:func:`Py_SetPythonHome`
+  * :c:func:`Py_SetStandardStreamEncoding`
+
+* Utilities:
+
+  * :c:func:`Py_DecodeLocale`
+
+* Memory allocators:
+
+  * :c:func:`PyMem_RawMalloc`
+  * :c:func:`PyMem_RawRealloc`
+  * :c:func:`PyMem_RawCalloc`
+  * :c:func:`PyMem_RawFree`
+  * :c:func:`PyMem_Malloc`
+  * :c:func:`PyMem_Realloc`
+  * :c:func:`PyMem_Calloc`
+  * :c:func:`PyMem_Free`
+  * :c:func:`PyObject_Malloc`
+  * :c:func:`PyObject_Realloc`
+  * :c:func:`PyObject_Calloc`
+  * :c:func:`PyObject_Free`
+
+
+.. note::
+
+   :c:func:`Py_GetPath`, :c:func:`Py_GetPrefix`, :c:func:`Py_GetExecPrefix` and
+   :c:func:`Py_GetProgramFullPath` and :c:func:`Py_GetPythonHome` **should not
+   be called** before :c:func:`Py_Initialize`.
+
+
+.. _global-conf-vars:
+
+Global configuration variables
+==============================
+
+Python has variables for the global configuration to control different features
+and options. By default, these flags are controlled by :ref:`command line
+options <using-on-interface-options>`.
+
+When a flag is set by an option, the value of the flag is the number of times
+that the option was set. For example, ``-b`` sets :c:data:`Py_BytesWarningFlag`
+to 1 and ``-bb`` sets :c:data:`Py_BytesWarningFlag` to 2.
+
+.. c:var:: Py_BytesWarningFlag
+
+   Issue a warning when comparing :class:`bytes` or :class:`bytearray` with
+   :class:`str` or :class:`bytes` with :class:`int`.  Issue an error if greater
+   or equal to ``2``.
+
+   Set by the :option:`-b` option.
+
+.. c:var:: Py_DebugFlag
+
+   Turn on parser debugging output (for wizards only, depending on compilation
```

Comment:
I see the word **wizards** and thought of magicians :) Do you mean something like setup wizard?
Or only a real wizard/magician can use this flag? 😅 
Maybe this is a common terminology and I'm just too new.. 🤷‍♀️

Reply:
I think Victor meant real [wizards](http://www.catb.org/jargon/html/W/wizard.html), like his and me. :tophat:

Reply:
Ok in that case I suggest something less ambiguous, like "for expert" or "for experienced core devs" only.
This sentence also missed a period (`.`) at the end.

Reply:
This wording already is used two times in the documentation. In particularly in the documentation for the `-d` option. This is very old documentation.

Reply:
In sum, this option is only for those who understand this jargon. :wink:

Reply:
I copied the doc from the -d option documentation.

Ok, I replaced "for wizards only" with "for expert only".

Reply:
I would keep the original wording. It is used also in the man page since the initial edition in 1994.